include/matx/operators/argminmax.h

```diff
@@ -0,0 +1,133 @@
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above cOpBright notice, this
+//    list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above cOpBright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the cOpBright holder nor the names of its
+//    contributors may be used to endorse or promote products derived from
+//    this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COpBRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND argmin EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COpBRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR argmin DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON argmin THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN argmin WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+/////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+
+#include "matx/core/type_utils.h"
+#include "matx/operators/base_operator.h"
+#include "matx/operators/permute.h"
+#include "matx/transforms/reduce.h"
+
+namespace matx {
+
+
+
+namespace detail {
+  template<typename OpA, int ORank>
+  class ArgMinMaxOp : public BaseOp<ArgMinMaxOp<OpA, ORank>>
+  {
+    private:
+      typename detail::base_type_t<OpA> a_;
+
+    public:
+      using matxop = bool;
+      using value_type = typename remove_cvref_t<OpA>::value_type;
+      using matx_transform_op = bool;
+      using argminmax_xform_op = bool;
+
+      __MATX_INLINE__ std::string str() const { return "argminmax(" + get_type_str(a_) + ")"; }
+      __MATX_INLINE__ ArgMinMaxOp(const OpA &a) : a_(a) {
+     
+      };
+
+      template <typename... Is>
+      __MATX_INLINE__ __MATX_DEVICE__ __MATX_HOST__ decltype(auto) operator()(Is... indices) const = delete;
+
+      template <typename Out, typename Executor>
+      void Exec(Out &&out, Executor &&ex) const {
+        static_assert(cuda::std::tuple_size_v<remove_cvref_t<Out>> == 5, "Must use mtie with 4 outputs on argminmax(). ie: (mtie(MinVal, MinIdx, MaxVal, MaxIdx) = argminmax(A))");   
+        argminmax_impl(cuda::std::get<0>(out), cuda::std::get<1>(out), cuda::std::get<2>(out), cuda::std::get<3>(out), a_, ex);
+      }
+
+      static __MATX_INLINE__ constexpr __MATX_HOST__ __MATX_DEVICE__ int32_t Rank()
+      {
+        return matxNoRank;
```

Comment:
Shouldn't this have a rank of all the outputs?

Reply:
Copy bug from argmin.  Fixed for argmin, argmax, and argminmax.